backend/data_preparation/dumper/fire_dumper.py

```diff
@@ -0,0 +1,438 @@
+    SQL_INSERT_FIRE_INTO_MERGED = 'INSERT INTO fire_merged(name, if_sequence, agency, state, id, start_time, ' \
+                                  'end_time, geom_full, geom_1e4, geom_1e3, geom_1e2, geom_center, max_area)' \
+                                  'Values (%s, %s, %s,%s, %s, %s,%s,%s,%s,%s,%s,%s,%s) ON CONFLICT (id) DO UPDATE SET ' \
+                                  'if_sequence = true, agency = excluded.agency, end_time = excluded.end_time, ' \
+                                  'geom_full = excluded.geom_full,geom_1e4 = Excluded.geom_1e4,geom_1e3 = Excluded.geom_1e3,' \
+                                  'geom_1e2 = Excluded.geom_1e2, geom_center = EXCLUDED.geom_center,max_area=EXCLUDED.max_area'
+    # SQL_INSERT_FIRE_INTO_MERGED: insert a new fire record into fire_merged
+    SQL_UPDATE_FIRE_INFO = 'UPDATE fire SET id = %s WHERE name = %s AND time >= %s::timestamp AND ' \
+                           'time <= %s::timestamp;'
+    # update fire info when: there are one fire in temp folder but records are from fireEvents with different name, one
+    # record will be seperated into two records
+    # so id in fire table need to be updated with a larger number
+
+    # code for requesting fire information
+    # retrieve all fire information
+    SQL_RETRIEVE_ALL_FIRES = 'SELECT year,state,name FROM fire_history'
+    # get the last id from fire_history table
+    SQL_GET_LASTEST_ID = 'SELECT MAX(id) FROM fire_history'
+    # get the recent fire whose last record is within 10 days of the current date
+    SQL_GET_LATEST_FIRE = 'SELECT a.id,h.year, h.state, h.name ' \
+                          'from (SELECT id FROM fire_merged Where abs(DATE_PART(\'day\', ' \
+                          'end_time - now())) < 10) a, fire_history h where h.id = a.id;'
+    # return the aggregated fire information with a given id
+    SQL_GET_LATEST_AGGREGATION = 'SELECT f.name, f.if_sequence,string_agg(distinct (f.agency), \', \'),f.state,f.id,' \
+                                 'min(f.time),Max(f.time), st_astext(st_union(st_makevalid(f.geom_full))) as geom_full,' \
+                                 'st_astext(st_union(st_makevalid(f.geom_1e4))),st_astext(st_union(st_makevalid(f.geom_1e3))),' \
+                                 'st_astext(st_union(st_makevalid(f.geom_1e2))),st_astext(st_centroid(st_union(st_makevalid(f.geom_center)))), ' \
+                                 'max(f.area) FROM (SELECT * FROM fire where id = {}) f Group by f.name, f.if_sequence, f.state, f.id'
+
+    @staticmethod
+    def _get_length_of_select_query_result(query: str) -> int:
+        """
+        Takes a SELECT query and returns the number of rows the query returns.
+        :param query: str.
+                e.g. 'SELECT year,state,name FROM fire_history'
+        :return: int
+        """
+        return sum(1 for _ in Connection.sql_execute(query))
+
+    @staticmethod
+    def _create_table(table_name: str):
```

Comment:
maybe change the name to `_create_table_if_not_exist` or something better explain the method.